.github/actions/integration-tests/action.yaml

```
@@ -19,6 +19,12 @@ inputs:
 runs:
   using: "composite"
   steps:
+    - name: Checkout to PR branch
+      shell: bash
+      if: ${{ github.event.type == 'PullRequestEvent' }}
+      run: |
+        git fetch origin pull/${{ github.event.pull_request.number }}/head:${{ github.head_ref }}
+        git checkout ${{ github.head_ref }}
```

If the head ref is e.g. `main` this will fail
```suggestion
        git fetch origin pull/${{ github.event.pull_request.number }}/head:PR-${{ github.event.pull_request.number }}
        git checkout PR-${{ github.event.pull_request.number }}
```